parlai/tasks/commonsenseqa/agents.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import FixedDialogTeacher
+from .build import build
+import os
+import json
+
+
+class CommonSenseQATeacher(FixedDialogTeacher):
+    """
+    CommonSenseQA is a multiple-choice Q-A dataset that relies on commonsense knowlegde
+    to predict correct answers. More information found at:
+
+    <https://www.tau-nlp.org/commonsenseqa>.
+    """
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+        self.id = 'commonsenseqa'
+        build(opt)
+        if shared is not None:
+            self.episodes = shared['episodes']
+        else:
+            self.episodes = self.setup_data()
+        self.reset()
+
+    def share(self):
+        shared = super().share()
+        shared['episodes'] = self.episodes
+        return shared
+
+    def setup_data(self):
+        jsons_path = os.path.join(self.opt['datapath'], 'CommonSenseQA')
+        dtype = self.opt['datatype']
+        if dtype.startswith('valid') or dtype.startswith('test'):
+            dpath = os.path.join(jsons_path, 'dev.jsonl')
+        elif dtype.startswith('train'):
+            dpath = os.path.join(jsons_path, 'train.jsonl')
+        else:
+            raise ValueError('Datatype not train, test, or valid')
+        episodes = []
+        with open(dpath) as f:
+            for line in f:
+                episodes.append(json.loads(line))
+        if dtype.startswith('valid'):
+            episodes = episodes[len(episodes) // 2 :]
+        elif dtype.startswith('test'):
```

Comment:
is this how the dataset is setup? It might be good to `warn_once` that there is no available test set, rather than splitting up the validation set (unless the second half of the dev set _is_ the test set)